src/main/java/Parser.java

```diff
@@ -0,0 +1,85 @@
+public class Parser {
+    public static void addOrCompleteTask(String input, Task[] tasks){
+        String[] inputData = input.split(" ");
+        String[] taskDetails;
+        switch (inputData[0]) {
+        case "done": {
+            int completedTask = Integer.parseInt(input.split(" ")[1]);
+            completedTask--;
+            tasks[completedTask].markAsDone();
+            System.out.println("The Force is with you! The following task has been marked as done:");
+            System.out.println(tasks[completedTask]);
+            System.out.println("You still have " + Task.getTaskCounter() + " tasks in your list!");
+            UI.showDivider();
+
+            break;
+            }
+        case "deadline": {
+            taskDetails = findTaskDetails(inputData);
+            tasks[Task.getTaskCounter()] = new Deadline(taskDetails[0], taskDetails[1]);
+            System.out.println("Roger Roger. The following task has been added:");
+            System.out.println(tasks[Task.getTaskCounter()]);
+            Task.incrementTaskCounter();
+            System.out.println("You now have " + Task.getTaskCounter() + " tasks to complete in your list!");
+            UI.showDivider();
```

Comment:
This highlighted block of code can be shift

Reply:
Good point! I've isolated them!